oss_src/sframe_query_engine/planning/optimizations/general_union_project_transforms.hpp

```diff
@@ -224,10 +224,16 @@ class opt_project_add_direct_source_tags : public opt_transform {
       auto nn = n->inputs[idx_1];
       
       if(nn->type == planner_node_type::SFRAME_SOURCE_NODE) {
-        input_mapping[i] = nn->any_p<sframe>("sframe").select_column(idx_2); 
+        auto sa = nn->any_p<sframe>("sframe").select_column(idx_2);
+        if (nn->p("begin_index") == 0 && nn->p("end_index") == sa->size()) {
```

Comment:
If the SOURCE node involves a sub-selected range (begin row, end row), than the input-mapping is insufficient information (since the input-mapping only contains column-id to the actual sarray, and not the range).